### PR TITLE
1043: Fixed search options

### PR DESCRIPTION
**Pull request by nlubisch**

### CODE REVIEW

File: htdocs/search.php [hunk 1]
```diff
@@ -1108,17 +1120,11 @@
         }
 
         // additional options
-        if (!isset($options['f_userowner'])) {
-            $options['f_userowner'] = '0';
-        }
-        if ($options['f_userowner'] != 0) {  // Ocprop
+        if ($options['f_userowner'] === 1) {  // Ocprop
```
Comment (Haehnchen) [inline htdocs/search.php:1123]: u should migrate to Symfony Request where possible. this strict type is at risk as an int is not guaranteed see https://www.w3schools.com/php/filter_validate_boolean.asp

https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/ParameterBag.php#L185

Comment (nlubisch) [reply htdocs/search.php:1123]: @Haehnchen Feel free to join our development team and start migrating legacy code to the upcoming symfony app ;)

Comment (following5) [reply htdocs/search.php:1123]: Agree that this strict typisation is problematic; the code becomes less robust and harder to verify. What is the purpose of this fix? What is the bug to be fixed?

**Comment by nlubisch** [comment]
Closed because outdated.

**Pull request closed without merging**